src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/CertificateUtil.kt

```diff
@@ -2,43 +2,50 @@ package ch.hevs.cloudio.cloud.apiutils
 
 import ch.hevs.cloudio.cloud.internalservice.CertificateAndPrivateKey
 import ch.hevs.cloudio.cloud.internalservice.CertificateFromKey
-import ch.hevs.cloudio.cloud.internalservice.UuidAndPublicKey
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.module.kotlin.KotlinModule
 import org.springframework.amqp.rabbit.core.RabbitTemplate
 import org.springframework.core.env.Environment
-import java.util.*
 
 object CertificateUtil{
 
     fun createCertificateAndKey(rabbitTemplate: RabbitTemplate, certificateAndKeyRequest: CertificateAndKeyRequest): CertificateAndPrivateKey{
-        val uuid = UUID.fromString(certificateAndKeyRequest.endpointUuid)
-
         val mapper = ObjectMapper().registerModule(KotlinModule())
 
         //set waiting time to infinite --> wait until the Certificate manager service turns on
         rabbitTemplate.setReplyTimeout(-1)
         val certificateAndPrivateKey = CertificateAndPrivateKey("","")
-        val uuidString = mapper.writeValueAsString(uuid)
+        val certificateAndKeyRequestString = mapper.writeValueAsString(certificateAndKeyRequest)
         val certificateFromUUID = rabbitTemplate.convertSendAndReceive("cloudio.service.internal",
-                "endpointKey-certificatePair", uuidString) as String?
+                "endpointKey-certificatePair", certificateAndKeyRequestString) as String?
         mapper.readerForUpdating(certificateAndPrivateKey).readValue(certificateFromUUID) as CertificateAndPrivateKey?
         //reset waiting time
         rabbitTemplate.setReplyTimeout(0)
 
         return certificateAndPrivateKey
     }
 
-    fun createCertificateFromKey(rabbitTemplate: RabbitTemplate, certificateFromKeyRequest: CertificateFromKeyRequest): CertificateFromKey{
-        val uuidAndPublicKey = UuidAndPublicKey(UUID.fromString(certificateFromKeyRequest.endpointUuid),
-                certificateFromKeyRequest.publicKey)
-
+    fun createCertificateAndKeyZip(rabbitTemplate: RabbitTemplate, certificateAndKeyZipRequest: CertificateAndKeyZipRequest): String?{
         val mapper = ObjectMapper().registerModule(KotlinModule())
+
+        //set waiting time to infinite --> wait until the Certificate manager service turns on
+        rabbitTemplate.setReplyTimeout(-1)
```

Comment:
No problem to use a relatively big timeout here, but infinite can block all worker threads on a node!